docs/guides/tutorials/chatgpt_bot.rst

```diff
@@ -0,0 +1,2285 @@
+
+.. lint-on
+
+In this tutorial we're going to build a documentation chatbot with
+`Next.js <https://nextjs.org/>`_, `OpenAI <https://openai.com/>`_, and EdgeDB.
+
+.. warning::
+
+    This project makes calls to OpenAI's APIs. At the time of publication, new
+    users are granted $5 in API credits to use within the first three months
+    after registration. Trials are granted per phone number, not per account.
+    If you exhaust your trial credits or if your three months lapse, you will
+    need to switch to a paid account in order to build the tutorial project.
+
+How it works
+============
+
+*tl;dr- Training a language model is hard, but using embeddings to give it
+access to information beyond what it's trained on is easy… so we will do that!
+Now,* :ref:`skip ahead to get started building <ref_guide_chatgpt_bot_start>`
+*or read on for more detail.*
+
+Our chatbot is backed by `OpenAI's ChatGPT <https://openai.com/blog/chatgpt>`_.
+ChatGPT is an advanced large language model (LLM) that uses machine learning
+algorithms to generate human-like responses based on the input it's given.
+
+There are two options when integrating ChatGPT and language models in general:
+fine-tuning the model or using `embeddings
+<https://platform.openai.com/docs/guides/embeddings/what-are-embeddings>`_.
+Fine-tuning produces the best result, but it needs more of everything: more
+money, more time, more resources, and more training data. That's why many
+people and businesses use embeddings instead to provide additional context to
+an existing language model.
+
+Embeddings are a way to convert words, phrases, or other types of data into a
+numerical form that a computer can do math with. All of this is built on top
+of the foundation of natural language processing (NLP) which allows computers
+to fake an understanding of human language. In the context of NLP, word
+embeddings are used to transform words into vectors. These vectors define a
+word's position in space where the computer sorts them based on their
+syntactic and semantic similarity. For instance, synonyms are closer to each
+other, and words that often appear in similar contexts are grouped together.
+
+When using embeddings we are not training the language model. Instead we're
+creating embedding vectors for every piece of documentation which will later
+help us find which piece of documentation likely answers a user's question.
+When a user asks a question, we create a new embedding for that question and
+compare it against the embeddings generated from our documentation to find the
+most similar embeddings. The answer is generated using the content that
+corresponds to these similar embeddings.
+
+With that out of the way, let's walk through how the pieces fit together.
+
+
+Implementation overview
+-----------------------
+
+The general implementation has two steps (which we'll also follow in the
+guide):
+
+1. create embeddings for each section using `OpenAI's embeddings API
```

Comment:
I think that we should not exclude the first part which is getting the docs files into the right format and splitting that into sections. This is crucial part of the whole thing. 

Also I'm not sure when you say the general implementation has 2 steps, do you think on creating and storing embeddings. In general, also the next 3 steps are part of the implementation.

Reply:
> I think that we should not exclude the first part which is getting the docs files into the right format and splitting that into sections. This is crucial part of the whole thing.

What do you mean by "getting the docs files into the right format?" Can you show me where in the example app this happens and where we split the sections? I think I have missed those parts. My understanding was we didn't need them for this app, so that's why I removed them here.

> Also I'm not sure when you say the general implementation has 2 steps, do you think on creating and storing embeddings. In general, also the next 3 steps are part of the implementation.

Let me work on rewording this. You're right that it isn't strictly accurate the way I've written it.

Reply:
Pushed a change to address your second concern.